internal/highway/highway.go

```diff
@@ -54,6 +55,9 @@ func NewHighway(ctx context.Context, opts ...config.Option) (*core.HighwayServer
 	s.Engine.POST("/v1/registry/whois/create", s.CreateWhoIs)
 	s.Engine.POST("/v1/registry/whois/update", s.UpdateWhoIs)
 	s.Engine.POST("/v1/registry/whois/deactivate", s.DeactivateWhoIs)
+	s.Engine.POST("/v1/registry/query/whois/:did", s.QueryWhoIs)
```

Comment:
Change POST to GET.

Change `/v1/registry/query/whois/:did` -> `/v1/registry/whois/:did`. The fact that it's a query is implied here if it's a GET request.